src/Surfnet/ServiceProviderDashboard/Application/CommandHandler/Entity/EntityChangeRequestCommandHandler.php

```diff
@@ -0,0 +1,213 @@
+use Surfnet\ServiceProviderDashboard\Application\Exception\EntityNotFoundException;
+use Surfnet\ServiceProviderDashboard\Application\Service\EntityServiceInterface;
+use Surfnet\ServiceProviderDashboard\Application\Service\TicketService;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\Constants;
+use Surfnet\ServiceProviderDashboard\Domain\Entity\ManageEntity;
+use Surfnet\ServiceProviderDashboard\Domain\Mailer\Mailer;
+use Surfnet\ServiceProviderDashboard\Domain\Repository\EntityChangeRequestRepository;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Ticket;
+use Surfnet\ServiceProviderDashboard\Infrastructure\DashboardBundle\Factory\MailMessageFactory;
+use Surfnet\ServiceProviderDashboard\Infrastructure\HttpClient\Exceptions\RuntimeException\PublishMetadataException;
+use Symfony\Component\HttpFoundation\Session\Flash\FlashBagInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class EntityChangeRequestCommandHandler implements CommandHandler
+{
+    /**
+     * @var EntityChangeRequestRepository
+     */
+    private $repository;
+    /**
+     * @var TicketService
+     */
+    private $ticketService;
+
+    /**
+     * @var FlashBagInterface
+     */
+    private $flashBag;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var string
+     */
+    private $issueType;
+
+    /**
+     * @var MailMessageFactory
+     */
+    private $mailFactory;
+
+    /**
+     * @var Mailer
+     */
+    private $mailer;
+
+    /**
+     * @var string
+     */
+    private $summaryTranslationKey;
+
+    /**
+     * @var string
+     */
+    private $descriptionTranslationKey;
+
+    /**
+     * @var EntityServiceInterface
+     */
+    private $entityService;
+
+    public function __construct(
+        EntityChangeRequestRepository $repository,
+        EntityServiceInterface $entityService,
+        TicketService $ticketService,
+        FlashBagInterface $flashBag,
+        MailMessageFactory $mailFactory,
+        Mailer $mailer,
+        LoggerInterface $logger,
+        string $issueType
+    ) {
+        Assert::stringNotEmpty($issueType, 'Please set "jira_issue_type_entity_change_request" in parameters.yml');
+        $this->repository = $repository;
+        $this->entityService = $entityService;
+        $this->ticketService = $ticketService;
+        $this->mailFactory = $mailFactory;
+        $this->mailer = $mailer;
+        $this->flashBag = $flashBag;
+        $this->logger = $logger;
+        $this->issueType = $issueType;
+        $this->summaryTranslationKey = 'entity.publish.change_request.ticket.summary';
+        $this->descriptionTranslationKey = 'entity.publish.change_request.ticket.description';
+    }
+
+    /**
+     * Creates an entity change request in Manage
+     * @SuppressWarnings(PHPMD.ElseExpression)
+     */
+    public function handle(PublishProductionCommandInterface $command)
```

Comment:
A new class with 2 ignores. Are these justified?
The handle function is quite big. Does it adhere to the SingleResponsibilityPrinciple?

Reply:
This one's a pickle. This command handler is tasked with many things indeed. Maybe this is a good time to extract some of its (shared) logic into services of their own.

Thanks for the suggestion! I will add this refactor to the #523 that branched of this work. That prevents me to rebase/rework this too much.